enterprise/internal/database/perms_store.go

```diff
@@ -2155,17 +2171,197 @@ FROM
+		} else {
+			if !authzParams.AuthzEnforceForSiteAdmins {
+				// include all site admins
+				permsQueryConditions = append(permsQueryConditions, sqlf.Sprintf("users.site_admin"))
+			}
+
+			// TODO(milan): need to update read path for new data model
+			// load user who has access via permissions syncing
+			permsQueryConditions = append(permsQueryConditions, sqlf.Sprintf(`
+				(
+					SELECT object_ids_ints @> INTSET(repo.id)
+					FROM user_permissions
+					WHERE
+						user_id = users.id
+				)
+			`))
+		}
+	}
+
+	where := []*sqlf.Query{sqlf.Sprintf("(%s)", sqlf.Join(permsQueryConditions, " OR "))}
+
+	paginationArgs := &defaultPaginationArgs
+
+	if args != nil {
+		if args.PaginationArgs != nil {
+			paginationArgs = args.PaginationArgs
+		}
+
+		if args.Query != "" {
+			pattern := "%" + args.Query + "%"
+			where = append(where, sqlf.Sprintf("(users.username ILIKE %s OR users.display_name ILIKE %s)", pattern, pattern))
+		}
+	}
+
+	pa := paginationArgs.SQL()
+	if pa.Where != nil {
+		where = append(where, pa.Where)
+	}
+
+	query := sqlf.Sprintf(usersPermissionsInfoQueryFmt, repoID, sqlf.Join(where, " AND "))
+	query = pa.AppendOrderToQuery(query)
+	query = pa.AppendLimitToQuery(query)
+
+	rows, err := s.Query(ctx, query)
+	if err != nil {
+		return nil, err
+	}
+
+	perms := make([]*RepoPermission, 0)
+	for rows.Next() {
+		var u types.User
+		var updatedAt time.Time
+		var displayName, avatarURL sql.NullString
+		u.DisplayName = displayName.String
+		u.AvatarURL = avatarURL.String
+
+		err := rows.Scan(
+			&u.ID,
+			&u.Username,
+			&displayName,
+			&avatarURL,
+			&u.CreatedAt,
+			&u.UpdatedAt,
+			&u.SiteAdmin,
+			&u.BuiltinAuth,
+			pq.Array(&u.Tags),
+			&u.InvalidatedSessionsAt,
+			&u.TosAccepted,
+			&u.Searchable,
+			&dbutil.NullTime{Time: &updatedAt},
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		reason := UserRepoPermissionReasonPermissionsSync
+		if unrestricted {
+			reason = UserRepoPermissionReasonUnrestricted
+			updatedAt = time.Time{}
+		} else if u.SiteAdmin {
+			reason = UserRepoPermissionReasonSiteAdmin
+			updatedAt = time.Time{}
+		}
+
+		perms = append(perms, &RepoPermission{User: &u, Reason: reason, UpdatedAt: updatedAt})
+	}
+
+	return perms, nil
+}
+
+const usersPermissionsInfoQueryFmt = `
+SELECT 
+	users.id, 
+	users.username, 
+	users.display_name, 
+	users.avatar_url, 
+	users.created_at, 
+	users.updated_at, 
+	users.site_admin, 
+	users.passwd IS NOT NULL, 
+	users.tags, 
+	users.invalidated_sessions_at, 
+	users.tos_accepted, 
+	users.searchable,
+	(	SELECT updated_at
+		FROM user_permissions
+		WHERE
+			user_id = users.id
+			AND object_ids_ints @> INTSET(repo.id)
+	) AS permissions_updated_at
+FROM users, repo
+WHERE
+	users.deleted_at IS NULL
+	AND repo.id = %d
+	AND %s
+`
+
+func (s *permsStore) isRepoUnrestricted(ctx context.Context, repoID api.RepoID, authzParams *database.AuthzQueryParameters) (*bool, error) {
```

Comment:
I took a look at the usage of this function, and we don't really need _ternary logic_ in here, because anyway `nil` bool is resolved to `false` on the call site. We might as well spare the call site from nil-checks and return a value `bool` instead of pointer here.

```suggestion
func (s *permsStore) isRepoUnrestricted(ctx context.Context, repoID api.RepoID, authzParams *database.AuthzQueryParameters) (bool, error) {
```